### PR TITLE
HW1

### PR DESCRIPTION
**Pull request by lysiakin**

### CODE REVIEW

File: src/Controller/Api/ApiController.php [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+     *
+     * @Route("/api/my-info", methods="GET", name="client_info")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function actionClientInfo(Request $request): Response
+    {
+        if (!$request->headers->get('Content-type: application/json')) {
```
Comment (anyt) [inline src/Controller/Api/ApiController.php:22]: Have you tested this check? it should not work.
Also, it make sense to move it to a separate private method, in order to reuse in all the controller actions and reduce the copy-paste.

File: src/Controller/Api/ApiController.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    public function actionIndex(Request $request): Response
+    {
+        if (!$request->headers->get('Content-type: application/json')) {
+            throw new UnsupportedMediaTypeHttpException();
+        }
+
+        $data = $this->getDataFromFile();
+        $responseData = $this->prepareData($request, $data);
+
+        return new Response(
```
Comment (anyt) [inline src/Controller/Api/ApiController.php:62]: It's better to use JsonResponse

File: src/Controller/Api/ApiController.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+     * @param array $data
+     *
+     * @return array
+     */
+    private function prepareData(Request $request, array $data): array
+    {
+        $responseData = [];
+
+        foreach ($data as $product) {
+            $productUri = $this->formatUri($product[0]);
```
Comment (anyt) [inline src/Controller/Api/ApiController.php:134]: Why not to use the router' GeneratorInterface?

File: src/Controller/Api/ApiController.php (hunk 2 above)
Comment (lysiakin) [reply src/Controller/Api/ApiController.php:62]: Forget about it, thanks

File: src/Controller/Api/ApiController.php (hunk 1 above)
Comment (lysiakin) [reply src/Controller/Api/ApiController.php:22]: But it works. This check is very simple. So i can't realize why it wouldnt work. 
Yeah, missed this point.

File: src/Controller/Api/ApiController.php (hunk 3 above)
Comment (lysiakin) [reply src/Controller/Api/ApiController.php:134]: I'll keep it on mind, thanks

**Pull request merged**